examples/i2cdev/default/main/main.c

```diff
@@ -6,9 +6,20 @@
 void task(void *ignore)
 {
     i2c_dev_t dev = { 0 };
+#if CONFIG_I2CDEV_USING_LEGACY_I2C
     dev.cfg.sda_io_num = CONFIG_EXAMPLE_I2C_MASTER_SDA;
     dev.cfg.scl_io_num = CONFIG_EXAMPLE_I2C_MASTER_SCL;
-#if HELPER_TARGET_IS_ESP32
+#else
+    dev.master_bus_config.sda_io_num = CONFIG_EXAMPLE_I2C_MASTER_SDA;
+    dev.master_bus_config.scl_io_num = CONFIG_EXAMPLE_I2C_MASTER_SCL;
+    dev.master_bus_config.i2c_port = -1;
+    dev.master_bus_config.clk_source = I2C_CLK_SRC_DEFAULT;
+    dev.master_bus_config.intr_priority = 0;
+    dev.master_bus_config.glitch_ignore_cnt = 7;
+    dev.master_bus_config.trans_queue_depth = 0;
+    dev.master_bus_config.flags.enable_internal_pullup = false;
```

Comment:
Using the i2c_dev component in this way locks the user/programmer into using this component for all I2C actions. I think this doesn't fit with the features' of the updated I2C driver (queuing). Wouldn't it be much more flexible for this component to store a pointer instead of possessing the bus (again)?

Reply:
It would be. What I came up is: create an abstraction header, "i2c_bus.h" for writing less code in most of simple cases, and pass a pointer (and whatever it needs to function) to "i2c_master.h". Users are free to choose either "i2c_bus.h" or their own.

Reply:
Sure, that would be nice